editor.planx.uk/src/@planx/components/Section/Public.tsx

```diff
@@ -153,71 +153,112 @@ export function SectionsOverviewList({
+  const getClickHandler = (sectionId: string, status: SectionStatusEnum) => {
+    if (showChange && status === SectionStatusEnum.Completed) {
+      return () => changeFirstAnswerInSection(sectionId);
+    }
+    if (
+      status === SectionStatusEnum.NeedsUpdated ||
+      status === SectionStatusEnum.ReadyToStart ||
+      status === SectionStatusEnum.ReadyToContinue
+    ) {
+      return () => nextQuestion();
+    }
+    return undefined;
+  };
+
+  const getTag = (section: SectionStatusEnum, sectionTitle: string) => {
+    const tagTypes: Record<SectionStatusEnum, TagType> = {
+      [SectionStatusEnum.NeedsUpdated]: TagType.Alert,
+      [SectionStatusEnum.ReadyToStart]: TagType.Active,
+      [SectionStatusEnum.ReadyToContinue]: TagType.Active,
+      [SectionStatusEnum.Started]: TagType.Notice,
+      [SectionStatusEnum.NotStarted]: TagType.Notice,
+      [SectionStatusEnum.Completed]: TagType.Success,
     };
     const tagType = tagTypes[section];
 
-    const onClick =
-      tagType == TagType.Alert || tagType == TagType.Active
-        ? () => nextQuestion()
-        : () => {}; // no-op
-
-    return (
-      <Tag tagType={tagType} onClick={onClick} sectionTitle={sectionTitle}>
-        {section}
-      </Tag>
-    );
+    return <Tag tagType={tagType}>{section}</Tag>;
   };
 
   return (
     <DescriptionList>
-      {Object.entries(sectionNodes).map(([sectionId, sectionNode]) => (
-        <SectionRow key={sectionId}>
-          <SectionTitle>
-            {showChange &&
-            sectionStatuses[sectionId] === SectionStatus.Completed ? (
-              <Link
-                onClick={() => changeFirstAnswerInSection(sectionId)}
-                component="button"
-              >
-                <Typography
-                  variant="subtitle1"
-                  component="h4"
-                  color="primary"
-                  align="left"
-                >
-                  <span style={visuallyHidden}>{`Change `}</span>
-                  <strong>{sectionNode.data.title}</strong>
-                </Typography>
-              </Link>
-            ) : (
-              <Typography variant="subtitle1" component="h4" color="inherit">
-                <strong>{sectionNode.data.title}</strong>
-              </Typography>
-            )}
-            <ReactMarkdownOrHtml
-              source={sectionNode.data.description}
-              openLinksOnNewTab
-            />
-          </SectionTitle>
-          <SectionState>
-            {getTag(sectionStatuses[sectionId], sectionNode.data.title)}
-          </SectionState>
-        </SectionRow>
-      ))}
+      {Object.entries(sectionNodes).map(([sectionId, sectionNode]) => {
+        const status = sectionStatuses[sectionId];
+        const clickHandler = getClickHandler(sectionId, status);
+        const isClickable = clickHandler !== undefined;
+        const statusId = `section-${sectionId}-status`;
+        const hintId = `section-${sectionId}-hint`;
```

Comment:
Generally I'd take this as a sign that we need a new component here. 

If we need to define these variables for the logic to work, it's worth defining them as their own component instead of increasing in-line complexity.